fs/operations/operations.go

```diff
@@ -1028,22 +1028,38 @@ func HashLister(ctx context.Context, ht hash.Type, outputBase64 bool, downloadFl
 				<-concurrencyControl
 				wg.Done()
 			}()
-			sum, err := hashSum(ctx, ht, downloadFlag, o)
+			sum, err := hashSum(ctx, ht, outputBase64, downloadFlag, o)
 			if err != nil {
 				fs.Errorf(o, "%v", fs.CountError(err))
 				return
 			}
-			if outputBase64 {
-				hexBytes, _ := hex.DecodeString(sum)
-				sum = base64.URLEncoding.EncodeToString(hexBytes)
-			}
 			syncFprintf(w, "%*s  %s\n", width, sum, o.Remote())
 		}()
 	})
 	wg.Wait()
 	return err
 }
 
+// HashStream produces hash from a stream
```

Comment:
I don't think HashStream is a good name for this function... I'd expect a function of that name to take a stream and return a hash, not print a hashsum compatible line to `w` :-)

It is a public function so should have a bit more docs. Something like

```go

// HashStream outputs a line compatible with md5sum to w based on the
// input stream in and the hash type ht passed in. If outputBase64 is
// set then the hash will be base64 instead of hexadecimal.

```

I'm struggling to think of a better name though... `HashSumStream` maybe?

Reply:
I see your point! But impossible to find a better name.. 

One reason was to use a name similar to the existing function in operations called `HashLister`, which also writes hashsum compatible lines.

There seem to be some mixing of `Hashsum` vs `HashSum`, same with `Checksum` vs `CheckSum`, unless there is a distinction there that I'm not aware of? I was considering your suggestion spelled as `HashsumStream`.. but ended up with your `HashSumStream`.